### PR TITLE
Reply in PR comments

### LINKED ISSUE #1795: Reply in PR comments
There are 2 buttons to **Reply** in PR comments. The one in the alert action doesn't do anything. 
Should I remove it? @rnystrom

### PR DESCRIPTION
**Pull request by jdisho**
This PR will close #1795 🎉

### CODE REVIEW

File: Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift [hunk 1]
```diff
@@ -200,4 +207,17 @@ final class PullRequestReviewCommentsViewController: MessageViewController,
         focusedReplyModel = reply
     }
 
+    // MARK: IssueCommentSectionControllerDelegate
+
+    func didSelectReply(to sectionController: IssueCommentSectionController, commentModel: IssueCommentModel) {
+        setMessageView(hidden: false, animated: true)
+        messageView.textView.becomeFirstResponder()
+        let quote = commentModel.rawMarkdown.substringUntilNewLine()
+        messageView.text = ">\(quote)\n\n@\(commentModel.details.login)"
```
Comment (rnystrom) [inline Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift]: Can we make more of this shared in a common function? Especially since the logic is all basically the same.

File: Classes/Issues/Comments/IssueCommentSectionController.swift [hunk 2]
```diff
@@ -441,9 +441,9 @@ final class IssueCommentSectionController:
             ToastManager.showGenericError()
             return
         }
-
+        
         let alertTitle = NSLocalizedString("%@'s comment", comment: "Used in an action sheet title, eg. \"Basthomas's comment\".")
-
+
```
Comment (rnystrom) [inline Classes/Issues/Comments/IssueCommentSectionController.swift]: Can we remove this? Make sure to have your Xcode settings:

screen shot 2018-05-08 at 10 42 10 pm

File: Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift [hunk 3]
```diff
@@ -29,7 +30,10 @@ final class PullRequestReviewCommentsViewController: MessageViewController,
         return f
     }()
 
-    init(model: IssueDetailsModel, client: GithubClient, autocomplete: IssueCommentAutocomplete) {
+    init(model: IssueDetailsModel,
```
Comment (rnystrom) [inline Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift]: nit: if we're updating the style, can you put the first param on its own line? Consistent w/ the rest of the app.

```swift
init(
  model: IssueDetailsModel,
  client: GithubClient,
  // etc
```

File: Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift [hunk 4]
```diff
@@ -164,11 +168,14 @@ final class PullRequestReviewCommentsViewController: MessageViewController,
...
-            client: client,
-            autocomplete: autocomplete
+        case is NSAttributedStringSizing:
+            return IssueTitleSectionController()
+        case is IssueCommentModel:
+            return IssueCommentSectionController(
+                model: model,
+                client: client,
+                autocomplete: autocomplete,
+                issueCommentDelegate: self
```
Comment (rnystrom) [inline Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift]: IMO it's better to separate random style updates into their own PR. Otherwise it makes reviewing these more difficult (not sure what is new code vs just style changes).

File: Classes/PullRequestReviews/String+Substring.swift [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+//
+//  Created by Joan Disho on 07.05.18.
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+
+extension String {
+    func substringUntilNewLine() -> String {
+        let substring = self.components(separatedBy: .newlines)[0]
```
Comment (rnystrom) [inline Classes/PullRequestReviews/String+Substring.swift]: Can we update this to use `.first` instead so it doesn't crash if there are no newlines? Also a unit test would be 👌

File: Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift (hunk 4 above)
Comment (jdisho) [reply Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift]: I don't understand this! 🙃

File: Classes/Issues/Comments/QuotedComment.swift [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import Foundation
+
+class QuotedComment {
+    private let feed: Feed
+    private let commentModel: IssueCommentModel
+
+    init(in feed: Feed, with commentModel: IssueCommentModel) {
```
Comment (BasThomas) [inline Classes/Issues/Comments/QuotedComment.swift]: I feel like we shouldn't rename these external parameters in this `init`; it makes it feel like a function, which may be unintended.

File: Classes/Issues/Comments/QuotedComment.swift [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    var quote: String {
+        let text = commentModel.rawMarkdown.substringUntilNewLine()
+        return ">\(text)\n\n@\(commentModel.details.login)"
+    }
+
+    func scrollInto() {
```
Comment (BasThomas) [inline Classes/Issues/Comments/QuotedComment.swift]: `scrollTo`*? Also feel like this shouldn't be the responsibility of this class.

File: FreetimeTests/ReplyCommentTests.swift [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    "<details>\n" +
+    "<summary>Bug Report Dump (Auto-generated)</summary>\n" +
+    "<pre>\n" +
+    "Version 1.20.0 (1525621207)\n" +
+    "Device: iPhone 6s Plus (iOS 11.4)\n" +
+    "TestFlight: true\n" +
+    "</pre>\n" +
+    "</details>"
+
+    let anotherTexttWithLessLines = "Couldn't find an open ticket for this, close if there is though!" +
```
Comment (BasThomas) [inline FreetimeTests/ReplyCommentTests.swift]: Nit: one too many `t`'s.

Comment (BasThomas) [reply FreetimeTests/ReplyCommentTests.swift]: Also, fewer* lines. 😬

File: FreetimeTests/ReplyCommentTests.swift [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+    func test_substringUntilNewLine() {
+        XCTAssertEqual(
+            textWithNewLines.substringUntilNewLine(),
+            "Couldn't find an open ticket for this, close if there is though! ...")
+
+        XCTAssertEqual(
+            anotherTexttWithLessLines.substringUntilNewLine(),
+            "Couldn't find an open ticket for this, close if there is though!In conjuction with apps like working copy it would be super helpful to open merge requests! ...")
+
+        XCTAssertEqual(emptyText.substringUntilNewLine(),"")
```
Comment (BasThomas) [inline FreetimeTests/ReplyCommentTests.swift]: Nit: space after the comma.

File: Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift (hunk 4 above)
Comment (BasThomas) [reply Classes/PullRequestReviews/PullRequestReviewCommentsViewController.swift]: He means the changes in whitespace and newlines.

File: Classes/Issues/Comments/QuotedComment.swift (hunk 7 above)
Comment (jdisho) [reply Classes/Issues/Comments/QuotedComment.swift]: I agree with you, it feels like it's not the responsibility of this class to scroll. 
However, since this line of code is repeated , I decided to do it like it is.

File: FreetimeTests/ReplyCommentTests.swift (hunk 8 above)
Comment (jdisho) [reply FreetimeTests/ReplyCommentTests.swift]: Hah nice one 😄

**Comment by jdisho** [comment]
Should we anything regarding this PR? It’s a bit annoying that the Reply button does nothing in PR comments. 🤷🏻‍♂️

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jdisho** [comment]
cc: @rnystrom 😇

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rnystrom** [comment]
Closing as its stale

**Comment by jdisho** [comment]
Actually this PR closes #1795 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jdisho** [comment]
Should we reopen this? Unless you think replying in PR comments is not necessary. @rnystrom 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Huddie** [comment]
@jdisho was this ready to merge or had conflicts?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by jdisho** [comment]
This was open for a long time, so I guess it had merge conflicts @Huddie

**Comment by Huddie** [comment]
I’d guess @rnystrom was doing some house cleaning, prob doesn’t mind the PR it just was so old that he closed it until the conflicts are resolved. 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rnystrom** [comment]
I'll reopen for @jdisho to handle conflicts, then we can revisit
